services/secondary-storage/server/src/main/java/org/apache/cloudstack/storage/resource/IpTablesHelper.java

```diff
@@ -0,0 +1,50 @@
+package org.apache.cloudstack.storage.resource;
+
+import com.cloud.utils.script.Script;
+import org.apache.cloudstack.storage.template.DownloadManagerImpl;
+import org.apache.log4j.Logger;
+
+public class IpTablesHelper {
+    public static final Logger LOGGER = Logger.getLogger(IpTablesHelper.class);
+
+    public static final String OUTPUT_CHAIN = "OUTPUT";
+    public static final String INPUT_CHAIN = "INPUT";
+    public static final String INSERT = " -I ";
+    public static final String APPEND = " -A ";
+
+    public static boolean needsAdding(String chain, String rule) {
+        Script command = new Script("/bin/bash", LOGGER);
+        command.add("-c");
+        command.add("iptables -C " + chain + " " + rule);
+
+        String r1 = command.execute();
+        boolean needsAdding = (r1 != null && r1.contains("iptables: Bad rule (does a matching rule exist in that chain?)."));
```

Comment:
What about checking for the exit value? In case the output of the command varies it can affect the method's behavior

Reply:
I've been testing a lot with both, will do one last round and delete whatever is not appropriate.

Reply:
I tested both methods and the result code method does not work.